cmd/cleanupexamples/main.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 Upbound Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// main package for the cleanupexamples tooling, the tool to remove
+// uptest-specific code from published examples on the marketplace
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+// removeAnnotations removes specific annotations from a Kubernetes
+// unstructured object. It looks for annotations with prefixes
+// "upjet.upbound.io/" and "uptest.upbound.io/" and removes
+// them if they are present.
+func removeAnnotations(u *unstructured.Unstructured) {
+	annotations := u.GetAnnotations()
+	if annotations != nil {
+		annotationsToRemove := []string{
+			"upjet.upbound.io/",
+			"uptest.upbound.io/",
+		}
+
+		for key := range annotations {
+			for _, prefix := range annotationsToRemove {
+				if strings.HasPrefix(key, prefix) {
+					delete(annotations, key)
+					break
+				}
+			}
+		}
+
+		if len(annotations) == 0 {
+			u.SetAnnotations(nil)
+		} else {
+			u.SetAnnotations(annotations)
+		}
+	}
```

Comment:
Suggestion: Consider whether the following organization is clearer. I removed the outer nil check, because the for loop and `len()` handle nil slices gracefully.

```suggestion
  annotationsToRemove := []string{
	  "upjet.upbound.io/",
	  "uptest.upbound.io/",
  }
  
  annotations := u.GetAnnotations()
  for key := range annotations {
	  for _, prefix := range annotationsToRemove {
		  if strings.HasPrefix(key, prefix) {
			  delete(annotations, key)
			  break
		  }
	  }
  }
  
  if len(annotations) == 0 {
	  u.SetAnnotations(nil)
  } else {
	  u.SetAnnotations(annotations)
  }
```